apps/src/blockly/addons/cdoBlockShadow.js

```diff
@@ -0,0 +1,141 @@
+// Helper functions for block shadowing logic. We use block shadowing for
+// our "mini toolbox" blocks, which are blocks that contain their own toolbox.
+// The blocks in the toolbox (or the blocks pulled from the toolbox)
+// need to shadow the image selected in the parent
+// block.
+
+// This object maps the block types that shadow parents to the parent block type they can shadow.
+// It also includes the index of the image to shadow in the parent's inputList.
+export const blockShadowingPairs = {
+  gamelab_clickedSpritePointer: {
```

Comment:
oh good call. I'll refactor it

Reply:
I ended up moving this map into a new file under `spritelab/blockly` because I was getting weird issues if it was in blocks, my guess is something to do with the keys matching block types. Not super sure what the issue was there, but putting it elsewhere seemed like a good idea anyway.